### PR TITLE
feat: add liveSyncTargetLatency option and deprecate liveSyncMinLatency and liveSyncMaxLatency options

### PR DESCRIPTION
**Pull request by gkatsev**
Follow-up from https://github.com/shaka-project/shaka-player/pull/6683 to add a target latency config and deprecate the existing min/max options.

### CODE REVIEW

File: lib/player.js [hunk 1]
```diff
@@ -3646,6 +3648,39 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
           'streaming.preferNativeHls instead.');
     }
 
+    // map liveSyncMinLatency and liveSyncMaxLatency to liveSyncTargetLatency
+    // if liveSyncTargetLatency isn't set.
```
Comment (gkatsev) [inline lib/player.js:3652]: This works for mapping the values to target latency but won't work well if each option is set individually.
If I don't delete the config option as part of deprecation, I could probably default to `this.config_.streaming.liveSyncMinLatency` etc, thoughts?

File: lib/player.js [hunk 2]
```diff
@@ -6321,44 +6356,43 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
...
         } else if (this.manifest_.serviceDescription.maxLatency != null) {
           liveSyncMaxLatency = this.manifest_.serviceDescription.maxLatency;
         }
-        liveSyncPlaybackRate =
-          this.manifest_.serviceDescription.maxPlaybackRate ||
-          this.config_.streaming.liveSyncPlaybackRate;
-      }
-    }
-
-    let liveSyncMinLatency;
```
Comment (gkatsev) [inline lib/player.js:6345]: this is mostly to consolidate into a single if block for both.

File: lib/player.js [hunk 3]
```diff
@@ -6321,44 +6356,43 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
...
     let liveSyncMaxLatency;
     let liveSyncPlaybackRate;
+    let liveSyncMinLatency;
+    let liveSyncMinPlaybackRate;
+    const liveSyncTargetLatencyTolerance =
+      this.config_.streaming.liveSyncTargetLatencyTolerance;
+
     if (this.config_.streaming.liveSync) {
-      liveSyncMaxLatency = this.config_.streaming.liveSyncMaxLatency;
+      if ('liveSyncTargetLatency' in this.config_.streaming) {
```
Comment (gkatsev) [inline lib/player.js:6365]: at this point, we probably have a target latency configured, but it felt correct to guard against it.

File: lib/player.js (hunk 1 above)
Comment (avelad) [reply lib/player.js:3652]: It's a risk we can take. I would leave it as is

File: lib/player.js (hunk 3 above)
Comment (avelad) [reply lib/player.js:6365]: This is not necessary, because our default config has it. Please remove this if.

File: lib/player.js [hunk 4]
```diff
@@ -3646,6 +3648,39 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
...
+      config['streaming']['liveSyncTargetLatencyTolerance'] = mid;
+    }
+    // Deprecate 'streaming.liveSyncMaxLatency' configuration.
+    if (config['streaming'] && 'liveSyncMaxLatency' in config['streaming']) {
+      shaka.Deprecate.deprecateFeature(5,
+          'streaming.liveSyncMaxLatency',
+          'Please Use streaming.liveSyncTargetLatency and ' +
+          'streaming.liveSyncTargetLatencyTolerance instead. ' +
+          'Or, set the values in your DASH manifest');
+      delete config['streaming']['liveSyncMaxLatency'];
```
Comment (avelad) [inline lib/player.js:3672]: Just like we have done with sourceBufferExtraFeatures (for example) you should remove the documentation from the extern, and you should also add all this information to `/docs/tutorials/upgrade.md`

File: lib/player.js [hunk 5]
```diff
@@ -3646,6 +3648,39 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
...
+    }
+    // Deprecate 'streaming.liveSyncMinLatency' configuration.
+    if (config['streaming'] && 'liveSyncMinLatency' in config['streaming']) {
+      shaka.Deprecate.deprecateFeature(5,
+          'streaming.liveSyncMinLatency',
+          'Please Use streaming.liveSyncTargetLatency and ' +
+          'streaming.liveSyncTargetLatencyTolerance instead. ' +
+          'Or, set the values in your DASH manifest');
+      delete config['streaming']['liveSyncMinLatency'];
```
Comment (avelad) [inline lib/player.js:3681]: Just like we have done with sourceBufferExtraFeatures (for example) you should remove the documentation from the extern, and you should also add all this information to `/docs/tutorials/upgrade.md`

**Review by avelad — CHANGES_REQUESTED** [review]
LGTM, there are only some minor changes

**Review by avelad — APPROVED** [review]

**Pull request merged**